src/main/java/uk/nhs/digital/nhsconnect/nhais/service/RecepProducerService.java

```diff
@@ -36,96 +41,108 @@ private String toEdifact(List<Segment> segments) {
-//    }
-//
-//    private RecepHeader mapToRecipInterchangeHeader(Interchange interchange) {
-//        var recepSender = interchange.getInterchangeHeader().getRecipient();
-//        var recepRecipient = interchange.getInterchangeHeader().getSender();
-//
-//        return new RecepHeader(recepSender, recepRecipient, interchange.getInterchangeHeader().getTranslationTime());
-//    }
-//
-//    private RecepMessageHeader mapToRecipMessageHeader(Interchange interchange) {
-//        return new RecepMessageHeader()
-//            .setSequenceNumber(interchange.getMessageHeader().getSequenceNumber());
-//    }
-//
-//    private RecepBeginningOfMessage mapToRecepBeginningOfMessage(Interchange interchange) {
-//        return new RecepBeginningOfMessage(interchange.getTranslationDateTime().getTimestamp());
-//    }
-//
-//    private RecepNameAndAddress mapToRecepNameAndAddress(Interchange interchange) {
-//        return new RecepNameAndAddress(interchange.getInterchangeHeader().getSender());
-//    }
-//
-//    private ReferenceMessageRecep mapToReferenceMessageRecep(Interchange interchange) {
-//        return new ReferenceMessageRecep(
-//            interchange.getMessageHeader().getSequenceNumber(), ReferenceMessageRecep.RecepCode.SUCCESS);
-//    }
-//
-//    private ReferenceInterchangeRecep mapToReferenceInterchangeRecep(Interchange interchange) {
-//        return new ReferenceInterchangeRecep(
-//            interchange.getInterchangeHeader().getSequenceNumber(),
-//            ReferenceInterchangeRecep.RecepCode.RECEIVED,
-//            interchange.getInterchangeTrailer().getNumberOfMessages());
-//    }
+    private List<Segment> mapEdifactToRecep(Interchange receivedInterchangeFromHa) throws EdifactValidationException {
+        List<Segment> recepMessageSegments = new ArrayList<>();
+
+        var recepInterchangeHeader = mapToRecepInterchangeHeader(receivedInterchangeFromHa);
+        var recepMessageHeader = mapToRecepMessageHeader();
+        recepMessageSegments.add(recepInterchangeHeader);
+        recepMessageSegments.add(recepMessageHeader);
+        var recepBeginningOfMessage = mapToRecepBeginningOfMessage();
+        recepMessageSegments.add(recepBeginningOfMessage);
+        recepMessageSegments.add(mapToRecepNameAndAddress(receivedInterchangeFromHa));
+        var recepTranslationDateTime = mapToRecepTranslationDateTime();
+        recepMessageSegments.add(recepTranslationDateTime);
+        recepMessageSegments.addAll(mapToReferenceMessageReceps(receivedInterchangeFromHa));
+        recepMessageSegments.add(mapToReferenceInterchangeRecep(receivedInterchangeFromHa));
+        var recepMessageTrailer = mapToRecepMessageTrailer();
+        var recepInterchangeTrailer = mapToRecepInterchangeTrailer(receivedInterchangeFromHa);
+        recepMessageSegments.add(recepMessageTrailer);
+        recepMessageSegments.add(recepInterchangeTrailer);
+
+        recepMessageSegments.forEach(Segment::preValidate);
```

Comment:
(Optional) I think that dropping `recep` prefix from everything would improve readability - it's `RecepProducerService` so it's obvious that all of that is recep related